### PR TITLE
Alerts: first version of invalid v3 blocks

### PR DESCRIPTION
**Pull request by harding**
Previews:
- http://dg0.dtrt.org/  (main page, alerts banner)
- http://dg0.dtrt.org/en/alert/2015-07-04-spv-mining (message first draft)

I'll merge this version as soon as it gets an ack from someone knowledgable about the situation, and then begin working on a longer and more detailed version.

### CODE REVIEW

**Comment by theymos** [comment]
The good chain has overtaken the bad chain now, so I'm not sure that this is necessary.

**Comment by luke-jr** [comment]
@theymos v2 miners can still make 1 or 2 block false-confirmation for SPV/old nodes, and as long as F2Pool is "SPV mining", they will extend it even longer. So it's not really resolved until at least F2Pool fixes their stuff.

**Comment by ABISprotocol** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

David,

Wasn't it supposed to be something like "make sure and update your SPV
wallet to the most current version immediately" or something?  Because
the way it is currently worded makes it sound like you have to switch
to Core and there's no other way.

Correct me if I'm wrong though

On 07/03/2015 11:08 PM, David A. Harding wrote:
> Previews:
> 
> * http://dg0.dtrt.org/ (main page, alerts banner) *
> http://dg0.dtrt.org/en/alert/2015-07-04-spv-mining (message first
> draft)
> 
> I'll merge this version as soon as it gets an ack from someone 
> knowledgable about the situation, and then begin working on a
> longer and more detailed version.
> 
> ----------------------------------------------------------------------
- --
>
> 
> 
> You can view, comment on, or merge this pull request online at:
> 
> https://github.com/bitcoin-dot-org/bitcoin.org/pull/933
> 
> 
> Commit Summary
> 
> * Alerts: first version of invalid v3 blocks
> 
> 
> File Changes
> 
> * *A* _alerts/2015-07-04-spv-mining.html 
> <https://github.com/bitcoin-dot-org/bitcoin.org/pull/933/files#diff-0>
>
> 
(37)
> 
> 
> Patch Links:
> 
> * https://github.com/bitcoin-dot-org/bitcoin.org/pull/933.patch *
> https://github.com/bitcoin-dot-org/bitcoin.org/pull/933.diff
> 
> — Reply to this email directly or view it on GitHub 
> <https://github.com/bitcoin-dot-org/bitcoin.org/pull/933>.
> 

- -- 
http://abis.io ~
"a protocol concept to enable decentralization
and expansion of a giving economy, and a new social good"
https://keybase.io/odinn
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1

iQEcBAEBAgAGBQJVl31SAAoJEGxwq/inSG8CUKYH/3075u9wF/YG/4eU0mrj2gnE
ZSUCTg1LMS5Z3c8IZAIjXSkFprdUJLUmZmrqmkBD8iJqVErUcfO6mtj0s9h4ElJh
jd+GxQUUJePn2tmWM8oAO1zxFM0E/snXBODXvsLnw0W0FnnH9xHngluSc6ZSKlAG
zEiOcGXavJp5p/F1DKBFMpRa6wn6WnQogaXo2I+W/11RRZmR2P5qeBjxT7uvS0N9
BuxqyIYgDQHXHKkgfP9Wbo5fByHknIRRz+/yZ5lnz2B4ArbuyIuXggrsDYXHCOhV
AoQqZfscCmy65irxfv8y4TAELrog0SUhiCE9/aHAUuvGbRUgzTkVDEqoYFDDj9M=
=g4Kv
-----END PGP SIGNATURE-----

**Comment by luke-jr** [comment]
@ABISprotocol BitcoinJ has not been fixed and remains vulnerable in all versions.

**Comment by harding** [comment]
@ABISprotocol no, upgrading your SPV wallet won't help; it's a fundamental problem with SPV wallets---they trust the longest chain, even if it's an invalid chain.

**Comment by theymos** [comment]
@luke-jr Good point. I'm OK with merging this, then. Though it might be less panic-inducing to tell people to wait for ~30 confirmations if they're using the affected wallet software instead of telling them to stop accepting transactions completely.

File: _alerts/2015-07-04-spv-mining.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+<h2 id="summary">Summary</h2>
+
+<p>Some miners are currently generating invalid blocks. <b>Bitcoin Core
+   0.9.4 and earlier</b> will accept these invalid blocks, making them
+   more vulnerable to double spends.  This <b>also applies to web wallets</b>
+   from companies that still use 0.9.4 or earlier nodes.</p>
+
+<p><b>Lightweight (SPV) wallets are also at increased risk,</b> and will
+      remain so until all miners have switched back to full
+      validation.</p>
```
Comment (petertodd) [inline _alerts/2015-07-04-spv-mining.html]: "and will remain so until all miners have switch back to full validation, and the remaining miners who have not upgraded to support BIP66 upgrade."

**Comment by harding** [comment]
@theymos changing it to 30 confirmations per discussion on IRC.  After that I'm going to push it to the site and start on a more detailed notice.

Any comments left here will be incorporated into the more detailed version, and I'll post here with the PR number for the more detailed version when it's available.

**Comment by saivann** [comment]
@harding When displaying a panic alert in red, I think it's very important to make it clear what is the risk, and what isn't. For instance, mention clearly that people's existing bitcoins are safe. For example:

## What you should do

### Miners

...

### Users

Your current bitcoins are safe. However if you are receiving a payment, there is a higher risk of double-spend fraud until the issue is fixed.

Bitcoin Core users should upgrade to Bitcoin Core 0.9.5 or later.

...

**Comment by ABISprotocol** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

So, you're telling me that everyone using an SPV wallet should just
upgrade to Core? I'd like to hear more people weigh in on that, it
sounds like you are saying "all is lost" for all SPV wallets forever,
abandon all hope all ye who enter here. A bit hasty.

I would say hold off on that merge and wait for more comments, please.

On 07/03/2015 11:30 PM, David A. Harding wrote:
> @ABISprotocol <https://github.com/ABISprotocol> no, upgrading your
> SPV wallet won't help; it's a fundamental problem with SPV
> wallets---they trust the longest chain, even if it's an invalid
> chain.
> 
> — Reply to this email directly or view it on GitHub 
> <https://github.com/bitcoin-dot-org/bitcoin.org/pull/933#issuecomment-
118468776>.
>
> 
- -- 
http://abis.io ~
"a protocol concept to enable decentralization
and expansion of a giving economy, and a new social good"
https://keybase.io/odinn
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1

iQEcBAEBAgAGBQJVl36mAAoJEGxwq/inSG8CEFkIAIHa7DvoCUWryWFQ0+6vzFjo
peoymhxi09IHe4qOtuPpNb8TJfp7UsxVILJDcQMoYY5OmG9zW223iNLMk1u06WQf
VMwPKgwNr0VZkPtcY/Wf8NWRzpcRQNxRf6B8Pzo+FG8lX0gm3yHYIsLSs1XkqYMF
08WSPmu8agm/tawA54ujgUNlZmQIjMEcGoSLJ2SpuDf8TYz72Kje1G5Y9y4W/77e
6WbuyPyOtcAObgN9V/OmFIwZkEPNEnNusBcgmyK4dXr24yl80MFKx2fvHAA9HWBp
5t3zpErd5Zl1tVnNf3Taq97IjHtjHh5UP12fossg+c1Ih6ArUgmLFAdZaaL6mw4=
=TRFd
-----END PGP SIGNATURE-----

**Comment by harding** [comment]
@saivann I like the "your current bitcoins are safe".  I'll do that.

**Comment by luke-jr** [comment]
@ABISprotocol Light "SPV" nodes are inherently vulnerable to attacks, and the malfunctioning miners are effectively performing those attacks.

**Comment by petertodd** [comment]
@harding +1

**Comment by ABISprotocol** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

How about someone go talk to F2Pool?  Not exactly problem solved, but
then at least, part of this would be resolved, then you wouldn't have
people screaming "run for the exits, y'all."

I see no need to tell people they need to bail on their SPV software
forevah when simply a phone call or bitmessage to F2Pool might
actually be what needs to be done...

just my three bits

On 07/03/2015 11:27 PM, Luke-Jr wrote:
> @theymos <https://github.com/theymos> v2 miners can still make 1 or
> 2 block false-confirmation for SPV/old nodes, and as long as F2Pool
> is "SPV mining", they will extend it even longer. So it's not
> really resolved until at least F2Pool fixes their stuff.
> 
> — Reply to this email directly or view it on GitHub 
> <https://github.com/bitcoin-dot-org/bitcoin.org/pull/933#issuecomment-
118468663>.
>
> 
- -- 
http://abis.io ~
"a protocol concept to enable decentralization
and expansion of a giving economy, and a new social good"
https://keybase.io/odinn
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1

iQEcBAEBAgAGBQJVl3/IAAoJEGxwq/inSG8CVXoH/iTUFCplwcH3LGvQek1DKivN
x7D81bykGSoJOAdcSLIUyanqqQru9txK3VjdVJ0A0Fcb6njmEsdtm3FtRUGaZRjj
YSLFpYFt7iwyGn9SVLKPbs4kcDEfMQ3wOg8uai7BfWEE3DbGxXwmx5Psa3rpp289
/k8KW9GYNQqE89rHv5+7GTdm2rEWDFMDwZn3MQTs6OxpI3mjBLkp4+Qz50eoVeW/
hXBOb0nxG+YdhMld93i5rcrKFf7l2Mi35HBOeEIT+hp+8Geg+bD4bFNNJVIGuXGb
i+yT8P/deTot1mH4NRLMGrIsEYqVDprOp0aga92CLSBDx1CXuhmXWAgPetIpOCw=
=pL18
-----END PGP SIGNATURE-----

**Comment by harding** [comment]
@saivann when this is over, we should setup a shared google doc between everyone here and the other core devs to make joint alert editing easier.

**Comment by luke-jr** [comment]
@ABISprotocol They have been informed, but are not acting responsibly (shutting down the pool until it is fixed). In any case, even after they fix it, there remains 5% of the former miners working on invalid blocks.

**Comment by harding** [comment]
New commit addressing feedback from @saivann and others.  Preview updated.

**Comment by ABISprotocol** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

For people who are basically hearing about this but are ok with it
all, but don't want to update to Core, they'll shut down their SPV
client and wait it out.  Some will update to Core.

But what will happen if people just keep using SPVs as is likely the
case?  People aren't going to just shut down all their SPV wallets and
go hustling over to Core.  That's kind of a large change for most
people using SPVs and in fact there are a lot of people who originally
used Core and switched to SPVs and have no plans on going back...

Point of clarity... I am assuming that this alert is intended for all
users and not just for users who are SPV mining... so if all you do is
have an SPV wallet and you use it for ordinary transactions... then
you still should be concerned.

So, what will happen?  Am I correct to assume that basically there is
just another fork (or actually, longest valid chain)?  At some point,
the "Longest valid chain is the valid chain, in the eyes of the
original Bitcoin client" (per aaaaaaaarrrrrgh on reddit), therefore,
can a person with a SPV client simply keep their SPV wallet off and
wait for this drama to end, and then when it all resolves, turn it
back on?

Curious to know.

On 07/03/2015 11:39 PM, Luke-Jr wrote:
> @ABISprotocol <https://github.com/ABISprotocol> Light "SPV" nodes
> are inherently vulnerable to attacks, and the malfunctioning miners
> are effectively performing those attacks.
> 
> — Reply to this email directly or view it on GitHub 
> <https://github.com/bitcoin-dot-org/bitcoin.org/pull/933#issuecomment-
118470998>.
>
> 
- -- 
http://abis.io ~
"a protocol concept to enable decentralization
and expansion of a giving economy, and a new social good"
https://keybase.io/odinn
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1

iQEcBAEBAgAGBQJVl4PQAAoJEGxwq/inSG8CgjwIAIX1yFUl5Dhc0Ww8if8vo1z4
Y8Yb58S3B0sPrKEtbKFQjwCHolY8DGp0JyWBA5YgtFlJaTJjtg+/xs9Af3TIuLmA
/fsIHzMG+BkHP8xM01ChALvt9xQHdxekG/oWWK+cOSkeRkXN+4uanhd+R3w2gy3G
TbhYkd/Yf3MaFrWxT8J4/Rfq4zOH+aop4lLy3X6tXCNQpxlMrGY8Q/ftB08SJwqf
Qr2Fn5TC4uf/W+XIy7xYqCT7/2peJeAYI0b8YVS3SskODyR4wvtv8bg7hgqXWF8P
0/vahS2SmSSKOsgXs8GkixMDv+iZS/cuja/zdS4jBAcRIJ/SaQu0L/Nl83PikGg=
=DMNn
-----END PGP SIGNATURE-----

**Comment by saivann** [comment]
@harding I think that's excellent and as soon as @luke-jr or @petertodd ACKs, this should be merged (further improvements can be made if needed).

**Comment by luke-jr** [comment]
@ABISprotocol Yes, ignoring the SPV wallet until the situation is over (which may be weeks) is a sound approach. SPV wallets might release an update in the meantime to filter out this particular problem.

P.S. Please stop quoting and PGP signing your comments. :(

**Comment by petertodd** [comment]
To be clear, for Bitcoin Core users running >=0.10.0 (or 0.9.5) the situation never happened.

**Comment by petertodd** [comment]
ACK

**Comment by harding** [comment]
@petertodd thanks for the ack.  Merging now.

Everyone: please post suggested refinements here and I'll incorporate them in the update.  Thanks!

**Comment by luke-jr** [comment]
2 more changes: 1) perhaps express uncertainty as to F2Pool's current state; 2) Eligius is misspelled

**Comment by saivann** [comment]
@harding s/transations/transactions/

**Comment by saivann** [comment]
@harding Note; You might want to wait for the alert to be displayed before pushing additional commits, because that will otherwise cancel current ongoing build and slow down the update.

**Comment by harding** [comment]
@saivann & @luke-jr pushed 6ec25511828c390f97556bac456dd26e10c92cd8 to master with typo fixes only.

@saivann I was justed telling Luke on IRC that we needed to wait in order to prevent restarting the build. :-)

**Comment by GandalfBitcoin** [comment]
AntPool seemed to be "SPV-mining" too, due to this block:
https://blockchain.info/block-height/363734

**Comment by harding** [comment]
@GandalfBitcoin it looks like Peter and Luke are discussing on IRC how probable it is that block indicates SPV mining.  I'll see what they conclude before updating.  Thanks :-)

**Comment by harding** [comment]
Pushed 756dc81e74f3a5f560d45cd76118d55e3551e07c to master moving f2pool to the good list and removing Antpool since there's some doubt as to what they're doing.  (I don't think we need to list every safe pool; just provide a few options.)

**Comment by saivann** [comment]
@harding The commit was on your branch only. I pushed it to master and added another commit to remove an extra HTML tag.

**Comment by harding** [comment]
@saivann oh, thanks!

**Pull request merged**